go/vt/vtgate/evalengine/comparisons.go

```diff
@@ -156,8 +171,34 @@ func executeComparison(lVal, rVal EvalResult) (int, error) {
 		// 		comparison of floating-point numbers."
 		//
 		//		https://dev.mysql.com/doc/refman/8.0/en/type-conversion.html
-		return compareNumeric(makeFloat(lVal), makeFloat(rVal))
+		comp, err = compareNumeric(makeFloat(lVal), makeFloat(rVal))
+		return comp, false, err
+	}
+}
+
+func foldSingleLenTuples(val EvalResult) EvalResult {
+	if val.typ == querypb.Type_TUPLE && len(val.tupleResults) == 1 {
+		val = val.tupleResults[0]
+	}
+	return val
+}
+
+func compareTuples(lVal EvalResult, rVal EvalResult) (int, bool, error) {
+	if len(lVal.tupleResults) != len(rVal.tupleResults) {
+		return 0, false, vterrors.NewErrorf(vtrpcpb.Code_INVALID_ARGUMENT, vterrors.OperandColumns, "Operand should contain %d column(s)", len(lVal.tupleResults))
 	}
+	hasSeenNull := false
+	for idx, lResult := range lVal.tupleResults {
+		rResult := rVal.tupleResults[idx]
+		res, isNull, err := nullSafeExecuteComparison(lResult, rResult)
+		if isNull {
+			hasSeenNull = true
+		}
+		if res != 0 || err != nil {
+			return res, false, err
+		}
+	}
+	return 0, hasSeenNull, nil
 }
```

Comment:
let's move this to `evalengine.go` with the other compare functions

Reply:
Sure